src/pamiq_vrchat/sensors/audio.py

```diff
@@ -97,3 +98,48 @@ def get_device_name_vrchat_is_outputting_to() -> str | None:
 
     logger.warning("Can not find speaker device VRChat.exe is outputting to.")
     return None
+
+
+class AudioLengthCompletionWrapper(Wrapper[AudioFrame, AudioFrame]):
+    """Wrapper class that can complete audio length using audio output
+    previously."""
+
+    def __init__(self, frame_size: int, reset_buffer_on_pause: bool = True) -> None:
+        """Initialize.
+
+        Args:
+            frame_size: Target num of samples.
+            reset_buffer_on_pause: If True, make self._buffer empty when pause.
+        """
+        super().__init__()
+        self._frame_size = frame_size
+        self._reset_buffer_on_pause = reset_buffer_on_pause
+        self._buffer = None
+
+    @override
+    def wrap(self, value: AudioFrame) -> AudioFrame:
+        """Reads a frame from the Soundcard and outputs audio by adjusting
+        length.
+
+        Args:
+            value: Audio obtained from read() of AudioSensor (shape is [AudioSensor._frame_size, channels]).
+        Returns:
+            Audio adjusted it's length (shape is [min(AudioSensor._frame_size, self._frame_size), channels]).
+        """
+        assert value.ndim == 2
```

Comment:
assertではなくValueErrorを使ってください。また docstringの `Raises`セクションを追加し、どんなときにエラーが起きるか書いておいてください。
```suggestion
        if value.ndim != 2:
            raise ValueError(...)
```